python/ray/experimental/state/custom_types.py

```diff
@@ -0,0 +1,89 @@
+from typing import Literal
+
+from ray.dashboard.memory_utils import ReferenceType
+from ray.core.generated.gcs_pb2 import (
+    ActorTableData,
+    PlacementGroupTableData,
+    GcsNodeInfo,
+)
+from ray.core.generated.common_pb2 import (
+    TaskType,
+    WorkerType,
+    WorkerExitType,
+    TaskStatus,
+)
+
+
+ACTOR_STATUS = [
```

Comment:
Don't really like how I did this, but I am not surer what's the better alternative that's not overengineering (like auto generating this file). Looks like we cannot access the grpc enum when generating a doc because they mock them out not to compile ray before creating a doc (ping me if you'd like to talk this in person)

Reply:
Is the purpose of this so that we could reference the enum values in doc? 
Could we circumvent it by not explicitly listing all the enum values for the associated status in the docstring? But ask users to defer to the docs of the sources (protobuf or when these enums are documented). I feel the datasource should actually be where the definition of these states live, rather than in the state API since IIUC, we are simply mirroring them.

Reply:
I see - I agree. After looking at the generated doc, I think documenting these states make sense.